android/src/main/java/cl/json/social/TargetChosenReceiver.java

```diff
@@ -0,0 +1,91 @@
+package cl.json.social;
+
+import android.annotation.TargetApi;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.IntentSender;
+import android.os.Build;
+
+import com.facebook.react.bridge.Callback;
+import com.facebook.react.bridge.ReactContext;
+
+/**
+ * Receiver to record the chosen component when sharing an Intent.
+ */
+public class TargetChosenReceiver extends BroadcastReceiver {
+    private static final String EXTRA_RECEIVER_TOKEN = "receiver_token";
+    private static final Object LOCK = new Object();
```

Comment:
Is this so the app can support multiple shares "in flight" at the same time, and when they come back (possibly to different receivers?) they will resolve correctly for the callbacks?

Is that something that actually can happen? It seems like the app can't have control back from the share dialog until something has been chosen, meaning the callback would happen before the app could be used to share again. If that is correct, then perhaps all the parallel-aware stuff here can fall away? Or is this something that is documented as possible and you actually saw?

Reply:
This is just to avoid getting a call from anywhere else in the code. If by some mistake any event gets triggered it may lead to false positives. This here takes care of that, so that we can achieve a result as intended.